common/src/com/microsoft/alm/common/utils/UrlHelper.java

```diff
@@ -111,15 +113,68 @@ public static String trimLeadingAndTrailingSeparators(final String uri) {
     public static boolean isVSO(final URI uri) {
         if (uri != null && uri.getHost() != null) {
             final String host = uri.getHost().toLowerCase();
-            if (StringUtils.endsWith(host, HOST_VSO) || StringUtils.endsWith(host, HOST_TFS_ALL_IN)) {
+            if (StringUtils.endsWith(host, HOST_VSO) ||
+                    StringUtils.endsWith(host, HOST_TFS_ALL_IN) ||
+                    UrlHelper.isOrganizationHost(host)) {
                 return true;
             }
         }
         return false;
     }
 
+    public static boolean isOrganizationUrl(final String url) {
+        try {
+            URI uri = createUri(url);
+            return isOrganizationURI(uri);
+        }
+        catch(IllegalArgumentException ex) {
+            logger.debug(url, ex);
+        }
+        return false;
+    }
+
+    public static boolean isOrganizationURI(final URI uri) {
+        if (uri != null && uri.getHost() != null) {
+            return isOrganizationHost(uri.getHost());
+        }
+        return false;
+    }
+
+    public static boolean isOrganizationHost(final String host) {
+       if (StringUtils.equalsIgnoreCase(host, HOST_AZURE) ||
```

Comment:
If you want this to work with devfabric, there would need to be additional cases here. (I am guessing that is not a goal right now)

Reply:
It's not, but we ought to enumerate the URL we are expecting to support.